### DISCUSSION TITLE
Phoniebox / MQTT / HomeAssistant

### DISCUSSION THREAD

**Discussion by eingemaischt** [MiczFlor/RPi-Jukebox-RFID#1353] [Q&A]
### What functionality would you like to see in your phoniebox?

I would like to add my Phonieboxes to my HomeAssistant. The MQTT-Plugin in the Phoniebox transmits data to the MQTT broker, but I don't know how to add it to my HomeAssistant bc I found no example config.

### How do you envision the feature to work from a users perspective?

A user would see an example config in the repository which he or she may use...;-)

**Comment by s-martin** [comment]
Did you check https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Smart-Home-remote-control-with-MQTT
?

**Reply by eingemaischt** [reply]
Yes, and there's not a example configuration for HomeAssistant - only possible endpoints. You wrote that you are using OpenHAB - maybe one of the readers here tried HomeAssistant and has a sample config?

**Reply by splitti** [reply]
Hey, maybe this helps:
[https://splittscheid.de/home-assistant-steuert-jetzt-die-phoniebox-per-mqtt/](https://splittscheid.de/home-assistant-steuert-jetzt-die-phoniebox-per-mqtt/)

**Reply by andreasbrett** [reply]
> Hey, maybe this helps: https://splittscheid.de/home-assistant-steuert-jetzt-die-phoniebox-per-mqtt/

Great post, thank you! I took your comment *"Das Script arbeitet leider so, dass ein leerer Wert übergeben werden muss und das auskommentieren nicht ausreicht."* and improved the script. It will now allow uncommenting unused parameters rather than having to set them to "". The [change/PR](https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1609) is quite fresh but might get merged to the next release 2.4.

**Comment by c0un7-z3r0** [comment]
@eingemaischt I know it is an old post but maybe this is helpful to someone. 
https://github.com/c0un7-z3r0/hass-phoniebox

I was trying to integrate Phoniebox as "native" media player entity into Home Assistant. Currently it is only available as custom component via [HACS](https://hacs.xyz/) and is limited by what Phoniebox communicates via MQTT. Not all settings are integrated yet but all basic media player functionalities are working.

**Reply by andreasbrett** [reply]
Awesome! Let me know if I can help, I'm the author of the MQTT "plugin" in Phoniebox.

**Reply by c0un7-z3r0** [reply]
Cool :D Thanks for the implementation of MQTT and sorry that I was "silent" so long. But there is indeed one thing I wanted to ask before I start debugging myself. The `phoniebox/attribute/duration` is always `00:00:00`. At least using the spotify version of phoniebox. Do you know why this might be?

**Reply by c0un7-z3r0** [reply]
I might have found the issue and opened a MR https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1787. Seemingly at some point `duration` got removed and changed to `time`. I adjusted the regex accordingly and it works now on my local machine.
